src/compiler/parser.ts

```diff
@@ -4873,15 +4877,22 @@ namespace ts {
             return decorators;
         }
 
-        function parseModifiers(): ModifiersArray {
+        function parseModifiers(whitelist?: SyntaxKind[]): ModifiersArray {
             let flags = 0;
             let modifiers: ModifiersArray;
             while (true) {
                 const modifierStart = scanner.getStartPos();
                 const modifierKind = token;
 
-                if (!parseAnyContextualModifier()) {
-                    break;
+                if (whitelist && whitelist.indexOf(token) >= 0) {

```

Comment:
Yes, I have already found it (https://github.com/Microsoft/TypeScript/pull/5559#commitcomment-14282963)